internal/controller/ingress/service_controller.go

```diff
@@ -86,9 +92,19 @@ func (r *ServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				}
 				return shouldHandleService(svc)
 			},
-		})
-		// TODO: Add watches for modulesets and traffic policies so we get updates
-
+		}).
+		// Watch modulesets for changes
+		Watches(
+			&ingressv1alpha1.NgrokModuleSet{},
+			handler.EnqueueRequestsFromMapFunc(r.findServicesForModuleSet),
+		).
```

Comment:
Since the moduleset also can reference a secret, I assume we'd also want to trigger a reconcile when a secret is updated? This is the problem with the https edge https://github.com/ngrok/kubernetes-ingress-controller/issues/177 which I assume is applicable here as well? If so, should we also watch the secret here, or should we just solve this problem across multiple controllers at a later time?

Reply:
The secret watching needs to happen in the HTTPSEdge,TLSEdge, and TCPEdge controllers. We'll fix that in a future PR